stdlib/resources/resources.lua

```diff
@@ -55,17 +55,17 @@ function Resources.get_resource_patch_at(surface, position)
     while not Queue.is_empty(search_queue) do
         local current_tile = Queue.pop_left(search_queue)
         local current_entity = Resources.get_resource_at(surface, current_tile)
+        local current_tile_index = Tile.get_index(current_tile)
+
+        table.insert(visited_tiles, current_tile_index)
```

Comment:
This should be either `table.insert(visited_tiles, current_tile_index, true)` or `visited_tiles[current_tile_index] = true`
Right now it inserts the tile index as a value